tests/monad.cpp

```diff
@@ -143,3 +226,52 @@ TEST_CASE("rsl::has_value") {
         CHECK(exp.has_value());
     }
 }
+
+TEST_CASE("Monadic pipe on values") {
+    auto const I = [](auto x) { return x; };
+    auto const K = [](auto x) { return [=](auto) { return x; }; };
+    auto const mul = [](auto x, auto y) { return x * y; };
+    auto const bind1 = [](auto fn, auto x) { return [=](auto y) { return fn(x, y); }; };
+    auto const three = [](auto) { return int{3}; };
+    auto const wrap = [](auto x) { return result_t<decltype(x)>{x}; };
+
+    SECTION("5I = 5") {
+        auto const result = int{5} | I;
+        CHECK(result == 5);
+    }
+    SECTION("5I(K3) = 3") {
+        auto const result = int{5} | I | K(3);
+        CHECK(result == 3);
+    }
+    SECTION("K73 = 7") {
+        auto const result = (int{7} | I | K)(3);
+        CHECK(result == 7);
+    }
+    SECTION("4MUL3 = 12") {
+        auto const result = int{4} | bind1(mul, 3);
+        CHECK(result == 12);
+    }
+    SECTION("Different types") {
+        auto const result = double{5} | three;
+        CHECK(result == 3);
+    }
+    SECTION("Pipe value into function returning expected") {
+        auto const result = double{5} | wrap;
+        REQUIRE(result.has_value());
+        CHECK(result.value() == Catch::Approx(5));
+    }
+    SECTION("Pipe value into function returning expected pipe to bind") {
+        auto const result = double{5} | wrap | multiply_3;
+        REQUIRE(result.has_value());
+        CHECK(result.value() == Catch::Approx(5 * 3));
+    }
+}
+
+TEST_CASE("range-v3 operator| still works") {
+    SECTION("Basic transform") {
+        auto v = {20, 10, 15};
+        auto r_inv = v | ranges::views::transform([](int x) { return 1.0 / x; });
+        auto val = 1.0 / ranges::accumulate(r_inv, 0.0);
+        CHECK(val == Catch::Approx(4.61538));
+    }
+}
```

Comment:
Why are you testing 3rd party code?

Reply:
so I can make sure code with ranges operator| still builds when you include the monad.hpp header, that is all.